engine/src/main/java/org/hibernate/search/query/engine/spi/EntityInfo.java

```diff
@@ -23,9 +22,11 @@
 
 	Object[] getProjection();
 
-	List<Integer> getIndexesOfThis();
+	Iterable<Integer> getIndexesOfThis();
```

Comment:
But `Iterable` is a great return type if you don't want to expose neither what kind of collection it is, nor mutators ;-)